examples/consumer_rebalance_example/consumer_rebalance_example.go

```diff
@@ -0,0 +1,302 @@
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("%% Created Consumer %v\n", c)
+
+	// Subscribe to topics, call the rebalanceCallback on assignment/revoke.
+	// The rebalanceCallback can be triggered from c.Poll() and c.Close().
+	if err = c.SubscribeTopics(topics, rebalanceCallback); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to subscribe to %v: %s\n", topics, err)
+		os.Exit(1)
+	}
+
+	run := true
+	for run {
+		select {
+		case sig := <-sigchan:
+			fmt.Printf("%% Caught signal %v: terminating\n", sig)
+			run = false
+		default:
+			ev := c.Poll(100)
+			if ev == nil {
+				continue
+			}
+
+			if err = processEvent(c, ev); err != nil {
+				fmt.Fprintf(
+					os.Stderr, "Failed to process event: %v: %s\n", ev, err)
+			}
+
+		}
+	}
+
+	fmt.Printf("%% Closing consumer\n")
+	c.Close()
+}
+
+// processEvent processes the message/error received from the kafka Consumer's
+// Poll() method.
+func processEvent(c *kafka.Consumer, ev kafka.Event) error {
+	switch e := ev.(type) {
+
+	case *kafka.Message:
+		fmt.Printf("%% Message on %s:\n%s\n", e.TopicPartition, string(e.Value))
+
+		// Handle manual commit since enable.auto.commit is unset.
+		if err := maybeCommit(c, e.TopicPartition); err != nil {
+			return err
+		}
+
+	case kafka.Error:
+		// Errors should generally be considered informational, the client
+		// will try to automatically recover.
+		fmt.Fprintf(os.Stderr, "%% Error: %v: %v\n",
+			e.Code(), e)
+
+	default:
+		fmt.Printf("Ignored %v\n", e)
+	}
+
+	return nil
+}
+
+// maybeCommit is called for each message we receive from a Kafka topic.
+// This method can be used to apply some arbitary logic/processing to the
+// offsets, write the offsets into some external storage, and finally, to
+// decide when we want to commit already-stored offsets into Kafka.
+func maybeCommit(c *kafka.Consumer, topicPartition kafka.TopicPartition) error {
+	// Commit the already-stored offsets to Kafka whenever the offset is divisible
+	// by 10, otherwise return early.
+	// This logic is completely arbitrary. We can use any other internal or
+	// external variables to decide when we commit the already-stored offsets.
+	if topicPartition.Offset%10 != 0 {
+		return nil
+	}
+
+	committedOffsets, err := c.Commit()
+
+	// ErrNoOffset occurs when there are no stored offsets to commit. This
+	// can happen if we haven't stored anything since the last commit.
+	// While this will never happen for this example since we call this method
+	// per-message, and thus, always have something to commit, the error
+	// handling is illustrative of how to handle it in cases we call Commit()
+	// in another way, for example, every N seconds.
+	if err != nil && err.(kafka.Error).Code() != kafka.ErrNoOffset {
+		return err
+	}
+
+	fmt.Printf("%% Committed offsets to Kafka: %v\n", committedOffsets)
+	return nil
+}
+
+// rebalanceCallback is called on each group rebalance to assign additional
```

Comment:
I think it's good to list down possible scenarios when rebalance happens. 

1. Consumer Group subscribes to new topics.
2. New partition added.

possible cases like this.

Reply:
Done in the new PR